// space_pirates_api/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "DefaultSpacePiratesAPI", {
  enumerable: true,
  get: function get() {
    return _DefaultSpacePiratesAPI["default"];
  }
});
Object.defineProperty(exports, "FriendReqSchema", {
  enumerable: true,
  get: function get() {
    return _FriendReqSchema["default"];
  }
});
Object.defineProperty(exports, "FriendSchema", {
  enumerable: true,
  get: function get() {
    return _FriendSchema["default"];
  }
});
Object.defineProperty(exports, "NewUserSchema", {
  enumerable: true,
  get: function get() {
    return _NewUserSchema["default"];
  }
});
Object.defineProperty(exports, "UpdateExperienceSchema", {
  enumerable: true,
  get: function get() {
    return _UpdateExperienceSchema["default"];
  }
});
Object.defineProperty(exports, "UserResSchema", {
  enumerable: true,
  get: function get() {
    return _UserResSchema["default"];
  }
});

var _ApiClient = _interopRequireDefault(require("./ApiClient"));

var _FriendReqSchema = _interopRequireDefault(require("./model/FriendReqSchema"));

var _FriendSchema = _interopRequireDefault(require("./model/FriendSchema"));

var _NewUserSchema = _interopRequireDefault(require("./model/NewUserSchema"));

var _UpdateExperienceSchema = _interopRequireDefault(require("./model/UpdateExperienceSchema"));

var _UserResSchema = _interopRequireDefault(require("./model/UserResSchema"));

var _DefaultSpacePiratesAPI = _interopRequireDefault(require("./api/DefaultSpacePiratesAPI"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
