// space_pirates_api/dist/api/DefaultSpacePiratesAPI.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _FriendReqSchema = _interopRequireDefault(require("../model/FriendReqSchema"));

var _FriendSchema = _interopRequireDefault(require("../model/FriendSchema"));

var _NewUserSchema = _interopRequireDefault(require("../model/NewUserSchema"));

var _UpdateExperienceSchema = _interopRequireDefault(require("../model/UpdateExperienceSchema"));

var _UserResSchema = _interopRequireDefault(require("../model/UserResSchema"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }

/**
* Default service.
* @module api/DefaultSpacePiratesAPI
* @version 1.0.0
*/
var DefaultSpacePiratesAPI = /*#__PURE__*/function () {
  /**
  * Constructs a new DefaultSpacePiratesAPI. 
  * @alias module:api/DefaultSpacePiratesAPI
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function DefaultSpacePiratesAPI(apiClient) {
    _classCallCheck(this, DefaultSpacePiratesAPI);

    this.apiClient = apiClient || _ApiClient["default"].instance;
  }
  /**
   * Callback function to receive the result of the acceptFriendReq operation.
   * @callback module:api/DefaultSpacePiratesAPI~acceptFriendReqCallback
   * @param {String} error Error message, if any.
   * @param data This operation does not return a value.
   * @param {String} response The complete HTTP response.
   */

  /**
   * Accept a friend request
   * @param {String} userId the user's UUID (unique)
   * @param {String} friendUserId the user's UUID (unique)
   * @param {module:api/DefaultSpacePiratesAPI~acceptFriendReqCallback} callback The callback function, accepting three arguments: error, data, response
   */


  _createClass(DefaultSpacePiratesAPI, [{
    key: "acceptFriendReq",
    value: function acceptFriendReq(userId, friendUserId, callback) {
      var postBody = null; // verify the required parameter 'userId' is set

      if (userId === undefined || userId === null) {
        throw new Error("Missing the required parameter 'userId' when calling acceptFriendReq");
      } // verify the required parameter 'friendUserId' is set


      if (friendUserId === undefined || friendUserId === null) {
        throw new Error("Missing the required parameter 'friendUserId' when calling acceptFriendReq");
      }

      var pathParams = {
        'userId': userId,
        'friendUserId': friendUserId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['sessionAuth'];
      var contentTypes = [];
      var accepts = [];
      var returnType = null;
      return this.apiClient.callApi('/users/{userId}/friendRequests/{friendUserId}', 'PATCH', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the createFriend operation.
     * @callback module:api/DefaultSpacePiratesAPI~createFriendCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * create a friend relation
     * @param {String} userId the user's UUID (unique)
     * @param {String} friendUserId the user's UUID (unique)
     * @param {module:api/DefaultSpacePiratesAPI~createFriendCallback} callback The callback function, accepting three arguments: error, data, response
     */

  }, {
    key: "createFriend",
    value: function createFriend(userId, friendUserId, callback) {
      var postBody = null; // verify the required parameter 'userId' is set

      if (userId === undefined || userId === null) {
        throw new Error("Missing the required parameter 'userId' when calling createFriend");
      } // verify the required parameter 'friendUserId' is set


      if (friendUserId === undefined || friendUserId === null) {
        throw new Error("Missing the required parameter 'friendUserId' when calling createFriend");
      }

      var pathParams = {
        'userId': userId,
        'friendUserId': friendUserId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['sessionAuth'];
      var contentTypes = [];
      var accepts = [];
      var returnType = null;
      return this.apiClient.callApi('/users/{userId}/friendRequests/{friendUserId}', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the createUser operation.
     * @callback module:api/DefaultSpacePiratesAPI~createUserCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Creates a new user.
     * @param {String} userId id of user to create
     * @param {module:model/NewUserSchema} newUserSchema 
     * @param {module:api/DefaultSpacePiratesAPI~createUserCallback} callback The callback function, accepting three arguments: error, data, response
     */

  }, {
    key: "createUser",
    value: function createUser(userId, newUserSchema, callback) {
      var postBody = newUserSchema; // verify the required parameter 'userId' is set

      if (userId === undefined || userId === null) {
        throw new Error("Missing the required parameter 'userId' when calling createUser");
      } // verify the required parameter 'newUserSchema' is set


      if (newUserSchema === undefined || newUserSchema === null) {
        throw new Error("Missing the required parameter 'newUserSchema' when calling createUser");
      }

      var pathParams = {
        'userId': userId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['sessionAuth'];
      var contentTypes = ['application/json'];
      var accepts = [];
      var returnType = null;
      return this.apiClient.callApi('/users/{userId}', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the deleteFriend operation.
     * @callback module:api/DefaultSpacePiratesAPI~deleteFriendCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * delete a friend
     * @param {String} userId the user's UUID (unique)
     * @param {String} friendId the user's UUID (unique)
     * @param {module:api/DefaultSpacePiratesAPI~deleteFriendCallback} callback The callback function, accepting three arguments: error, data, response
     */

  }, {
    key: "deleteFriend",
    value: function deleteFriend(userId, friendId, callback) {
      var postBody = null; // verify the required parameter 'userId' is set

      if (userId === undefined || userId === null) {
        throw new Error("Missing the required parameter 'userId' when calling deleteFriend");
      } // verify the required parameter 'friendId' is set


      if (friendId === undefined || friendId === null) {
        throw new Error("Missing the required parameter 'friendId' when calling deleteFriend");
      }

      var pathParams = {
        'userId': userId,
        'friendId': friendId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['sessionAuth'];
      var contentTypes = [];
      var accepts = [];
      var returnType = null;
      return this.apiClient.callApi('/users/{userId}/friends/{friendId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the deleteUser operation.
     * @callback module:api/DefaultSpacePiratesAPI~deleteUserCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Delete a user
     * @param {String} userId 
     * @param {module:api/DefaultSpacePiratesAPI~deleteUserCallback} callback The callback function, accepting three arguments: error, data, response
     */

  }, {
    key: "deleteUser",
    value: function deleteUser(userId, callback) {
      var postBody = null; // verify the required parameter 'userId' is set

      if (userId === undefined || userId === null) {
        throw new Error("Missing the required parameter 'userId' when calling deleteUser");
      }

      var pathParams = {
        'userId': userId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['sessionAuth'];
      var contentTypes = [];
      var accepts = [];
      var returnType = null;
      return this.apiClient.callApi('/users/{userId}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the getUser operation.
     * @callback module:api/DefaultSpacePiratesAPI~getUserCallback
     * @param {String} error Error message, if any.
     * @param {module:model/UserResSchema} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Returns a user based on a single ID
     * @param {String} userId id of user to fetch
     * @param {module:api/DefaultSpacePiratesAPI~getUserCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/UserResSchema}
     */

  }, {
    key: "getUser",
    value: function getUser(userId, callback) {
      var postBody = null; // verify the required parameter 'userId' is set

      if (userId === undefined || userId === null) {
        throw new Error("Missing the required parameter 'userId' when calling getUser");
      }

      var pathParams = {
        'userId': userId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['sessionAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _UserResSchema["default"];
      return this.apiClient.callApi('/users/{userId}', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the readUserFriendRequests operation.
     * @callback module:api/DefaultSpacePiratesAPI~readUserFriendRequestsCallback
     * @param {String} error Error message, if any.
     * @param {Array.<module:model/FriendReqSchema>} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * read a users friend requests
     * @param {String} userId the user's UUID (unique)
     * @param {module:api/DefaultSpacePiratesAPI~readUserFriendRequestsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link Array.<module:model/FriendReqSchema>}
     */

  }, {
    key: "readUserFriendRequests",
    value: function readUserFriendRequests(userId, callback) {
      var postBody = null; // verify the required parameter 'userId' is set

      if (userId === undefined || userId === null) {
        throw new Error("Missing the required parameter 'userId' when calling readUserFriendRequests");
      }

      var pathParams = {
        'userId': userId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['sessionAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = [_FriendReqSchema["default"]];
      return this.apiClient.callApi('/users/{userId}/friendRequests', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the readUserFriends operation.
     * @callback module:api/DefaultSpacePiratesAPI~readUserFriendsCallback
     * @param {String} error Error message, if any.
     * @param {Array.<module:model/FriendSchema>} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * read a users friends
     * @param {String} userId the user's UUID (unique)
     * @param {module:api/DefaultSpacePiratesAPI~readUserFriendsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link Array.<module:model/FriendSchema>}
     */

  }, {
    key: "readUserFriends",
    value: function readUserFriends(userId, callback) {
      var postBody = null; // verify the required parameter 'userId' is set

      if (userId === undefined || userId === null) {
        throw new Error("Missing the required parameter 'userId' when calling readUserFriends");
      }

      var pathParams = {
        'userId': userId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['sessionAuth'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = [_FriendSchema["default"]];
      return this.apiClient.callApi('/users/{userId}/friends', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
    /**
     * Callback function to receive the result of the updateUserExperience operation.
     * @callback module:api/DefaultSpacePiratesAPI~updateUserExperienceCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Updates user experience
     * @param {String} userId the user's UUID (unique)
     * @param {module:model/UpdateExperienceSchema} updateExperienceSchema 
     * @param {module:api/DefaultSpacePiratesAPI~updateUserExperienceCallback} callback The callback function, accepting three arguments: error, data, response
     */

  }, {
    key: "updateUserExperience",
    value: function updateUserExperience(userId, updateExperienceSchema, callback) {
      var postBody = updateExperienceSchema; // verify the required parameter 'userId' is set

      if (userId === undefined || userId === null) {
        throw new Error("Missing the required parameter 'userId' when calling updateUserExperience");
      } // verify the required parameter 'updateExperienceSchema' is set


      if (updateExperienceSchema === undefined || updateExperienceSchema === null) {
        throw new Error("Missing the required parameter 'updateExperienceSchema' when calling updateUserExperience");
      }

      var pathParams = {
        'userId': userId
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['sigAuth'];
      var contentTypes = ['application/json'];
      var accepts = [];
      var returnType = null;
      return this.apiClient.callApi('/users/{userId}/experience', 'PATCH', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);

  return DefaultSpacePiratesAPI;
}();

exports["default"] = DefaultSpacePiratesAPI;
